src/Microsoft.AspNetCore.Rewrite/RewriteOptionsExtensions.cs

```diff
@@ -12,9 +13,36 @@ namespace Microsoft.AspNetCore.Rewrite
     public static class RewriteOptionsExtensions
     {
         /// <summary>
+        /// Set the default path base used to find config files
+        /// </summary>
+        /// <remarks>
+        /// This should be called before using
+        /// <see cref="ApacheModRewriteOptionsExtensions.AddApacheModRewrite(RewriteOptions, string)"/> or
+        /// <see cref="IISUrlRewriteOptionsExtensions.AddIISUrlRewrite(RewriteOptions, string)"/>.
+        /// </remarks>
+        /// <param name="options">The <see cref="RewriteOptions"/></param>
+        /// <param name="pathBase">The root path to use</param>
+        /// <returns></returns>
+        public static RewriteOptions SetConfigFileBasePath(this RewriteOptions options, string pathBase)
```

Comment:
IMO this should hang off an `IRewriteOptionsBuilder` API. This separates builder state from middleware config